### PR TITLE
New Post - Building a FAB component

### LINKED ISSUE #8537: [A11y][CSS] Building a floating action button (FAB) component
* Proposed title: Building a floating action button (FAB) component
* Proposed by: argyle
* Related areas: Accessibility,CSS
* Content type: Explanation (https://documentation.divio.com/explanation/)
* [Draft](https://docs.google.com/document/d/1_2uEB5dA1QeFnnQES3ga_HbAK9AXOIegVhcyYNR4KmU/edit#heading=h.9qj860fhfh8o)
* Content summary: FABs are a common component for primary actions and they're easy to mess up the accessibility, follow this and be on a safer track.

### PR DESCRIPTION
**Pull request by argyleink**
Fixes #8537

To Do:
- [x] final reviews
- [x] update publish date before merge

### CODE REVIEW

**Comment by argyleink** [comment]
this is still waiting for review 🙂

File: src/site/content/en/blog/building-a-fab-component/index.md [hunk 1]
```diff
@@ -0,0 +1,445 @@
+---
+layout: post
+title: Building a floating action button (FAB) component
+subhead: A foundational overview of how to build color-adaptive, responsive, and accessible FAB components.
+authors:
+  - adamargyle
+description: A foundational overview of how to build color-adaptive, responsive, and accessible FAB components.
+date: 2022-09-21
```
Comment (rachelandrew) [inline src/site/content/en/blog/building-a-fab-component/index.md:8]:
```suggestion
date: 2022-10-06
```

File: src/site/content/en/blog/building-a-fab-component/index.md [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+
+### FAB container
+
+This element can be a regular `<div>` but let's do our unsighted users a favor
+and tag it with some helpful attributes to explain the purpose and contents of
+this container.
+
+#### FABs markup
+
+Start with a ".fabs" class for CSS to hook into for style, then add
```
Comment (rachelandrew) [inline src/site/content/en/blog/building-a-fab-component/index.md:64]:
```suggestion
Start with a `.fabs` class for CSS to hook into for style, then add
```

File: src/site/content/en/blog/building-a-fab-component/index.md [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+[`column-reverse`](https://developer.mozilla.org/docs/Web/CSS/flex-direction).
+This stacks the children on top of each other (column) and also reverses their
+visual order. This has the effect of making the first focusable element the
+bottom element instead of the top, which would be where focus goes normally per
+the HTML document. Reversing the visual order unites the experience for sighted
+users and keyboard users, as the styling of the primary action as larger than
+the mini buttons indicates to sighted users that it's a primary action, and
+keyboard users will focus it as the first item in the source.
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/PkRWiv4mtfCDXujti7lL.png", alt="ALT_TEXT_HERE", width="800", height="501" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/building-a-fab-component/index.md:110]: This needs alt text.

File: src/site/content/en/blog/building-a-fab-component/index.md [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+Thanks to usage of custom properties, the only change needed is an adjustment to
+the `--_size` variable.
+
+```css
+.fab.mini {
+  --_size: 1.25rem;
+}
+```
+
+{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/bYhvGgYmV0OiztY7j58r.png", alt="ALT_TEXT_HERE", width="664", height="410" %}
```
Comment (rachelandrew) [inline src/site/content/en/blog/building-a-fab-component/index.md:290]: also missing alt text

**Review by rachelandrew — APPROVED** [review]
Happy to publish once the alt text is updated :)

File: src/site/content/en/blog/building-a-fab-component/index.md (hunk 3 above)
Comment (argyleink) [inline src/site/content/en/blog/building-a-fab-component/index.md:110]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/PkRWiv4mtfCDXujti7lL.png", alt="Two fab buttons are shown with DevTools overlaying their grid layout. Shows the gap between them with a striped pattern and also shows their computed height and width.", width="800", height="501" %}
```

File: src/site/content/en/blog/building-a-fab-component/index.md (hunk 4 above)
Comment (argyleink) [inline src/site/content/en/blog/building-a-fab-component/index.md:290]:
```suggestion
{% Img src="image/vS06HQ1YTsbMKSFTIPl2iogUQP73/bYhvGgYmV0OiztY7j58r.png", alt="A screenshot of the two fab buttons stacked and the top button is smaller than the one on the bottom.", width="664", height="410" %}
```

Comment (argyleink) [reply src/site/content/en/blog/building-a-fab-component/index.md:290]: ty, got em!

**Comment by argyleink** [comment]
> Happy to publish once the alt text is updated :)

all done! thanks a bunch!

**Pull request merged**